cdc/entry/mounter.go

```diff
@@ -126,60 +114,31 @@ func (m *mounterImpl) Run(ctx context.Context) error {
 		select {
 		// ctx.Done returns when parent ctx done or error occurs in errg.
 		case <-ctx.Done():
-			return errg.Wait()
+			return ctx.Err()
 		case <-timer.C:
-			chSize := 0
-			for _, ch := range m.rawRowChangedChs {
-				chSize += len(ch)
-			}
-			metricMounterInputChanSize.Set(float64(chSize))
 			timer.Reset(flushMetricsInterval)
 		}
 	}
 }
 
-func (m *mounterImpl) codecWorker(ctx context.Context, index int) error {
-	changefeedID := util.ChangefeedIDFromCtx(ctx)
-	metricMountDuration := mountDuration.WithLabelValues(changefeedID)
-	metricTotalRows := totalRowsCountGauge.WithLabelValues(changefeedID)
-	defer func() {
-		mountDuration.DeleteLabelValues(changefeedID)
-		totalRowsCountGauge.DeleteLabelValues(changefeedID)
-	}()
-
-	for {
-		var pEvent *model.PolymorphicEvent
-		select {
-		case <-ctx.Done():
-			return errors.Trace(ctx.Err())
-		case pEvent = <-m.rawRowChangedChs[index]:
-		}
-		if pEvent.RawKV.OpType == model.OpTypeResolved {
-			pEvent.PrepareFinished()
-			continue
-		}
-		startTime := time.Now()
-		rowEvent, err := m.unmarshalAndMountRowChanged(ctx, pEvent.RawKV)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		pEvent.Row = rowEvent
-		pEvent.RawKV.Value = nil
-		pEvent.RawKV.OldValue = nil
-		pEvent.PrepareFinished()
-		metricMountDuration.Observe(time.Since(startTime).Seconds())
-		metricTotalRows.Inc()
-	}
-}
-
-func (m *mounterImpl) AddEntry(ctx context.Context, event *model.PolymorphicEvent) error {
-	index := atomic.AddInt64(&m.index, 1) % int64(m.workerNum)
-	select {
-	case <-ctx.Done():
-		return ctx.Err()
-	case m.rawRowChangedChs[index] <- event:
+func (m *mounterImpl) DecodeEvent(ctx context.Context, pEvent *model.PolymorphicEvent) error {
+	m.metricTotalRows.Inc()
+	if pEvent.RawKV.OpType == model.OpTypeResolved {
 		return nil
 	}
+	start := time.Now()
+	rowEvent, err := m.unmarshalAndMountRowChanged(ctx, pEvent.RawKV)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	pEvent.Row = rowEvent
+	pEvent.RawKV.Value = nil
+	pEvent.RawKV.OldValue = nil
+	duration := time.Since(start)
+	if duration > time.Second {
```

Comment:
Could this condition produce misleading metrics results?

Reply:
yes, but according to this PR: https://github.com/pingcap/tiflow/pull/4585,  it's better to reduce some metrics overhead.